<a href="https://colab.research.google.com/github/String-ru/AIFFEL_quest_cr/blob/main/Datathon/1st_datathon_0218_3_30PM_1st_assumption.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#1st Assumption

#특정 배우는 특정 장르에서 더 높은 선호도를 보일까?

(장르에 대한 배우 개인의 선호도X)

[여러 장르에 출연하는 배우에 대한 시청자의 선호도O]

**특정 배우의 특정 장르에 대한 출연을 선호하는 경향이 시청자들 사이에 존재하는가?**

#개요

1. 배우별 장르 출연 횟수 세야 함

2. 장르별 시청자 선호도 파악 (imdb data set)

3. 배우와 장르의 관계파악 (특정 배우가 주로 출연하는 장르에서 시청자의 선호도가 높은지 여부 파악)

4. 그 관계 -> 시각화 & 통계적 분석

#Data 불러오기

In [1]:
# 1️⃣ Google Drive 마운트
from google.colab import drive
drive.mount('/content/drive')

# 2️⃣ 필요한 라이브러리 불러오기
import os
import pandas as pd

# 3️⃣ 데이터 폴더 경로 설정
data_folder = "/content/drive/MyDrive/1st_datathon"  # 폴더 경로

# 4️⃣ 폴더 내의 파일 목록 확인
file_list = os.listdir(data_folder)
print("📂 데이터 폴더 내 파일 목록:", file_list)

# 5️⃣ CSV 파일 경로 설정
disney_file = os.path.join(data_folder, "disney_plus_titles.csv")
amazon_file = os.path.join(data_folder, "amazon_prime_titles.csv")
netflix_file = os.path.join(data_folder, "netflix_titles.csv")

# 6️⃣ pandas로 CSV 파일 로드
disney_df = pd.read_csv(disney_file)
amazon_df = pd.read_csv(amazon_file)
netflix_df = pd.read_csv(netflix_file)

# 7️⃣ 데이터 크기 확인
print("✅ Disney+ 데이터:", disney_df.shape)
print("✅ Amazon Prime 데이터:", amazon_df.shape)
print("✅ Netflix 데이터:", netflix_df.shape)

Mounted at /content/drive
📂 데이터 폴더 내 파일 목록: ['disney_plus_titles.csv', 'amazon_prime_titles.csv', 'netflix_titles.csv']
✅ Disney+ 데이터: (1450, 12)
✅ Amazon Prime 데이터: (9668, 12)
✅ Netflix 데이터: (8807, 12)


=> 각각 1450 / 9668 / 8807개의 항목(영화나 TV쇼 etc.)에 대한 12개의 정보(column) 존재 확인.

#column 살짝 수정

In [2]:
# 컬럼 이름을 소문자로 통일
netflix_df.columns = netflix_df.columns.str.lower()
amazon_df.columns = amazon_df.columns.str.lower()
disney_df.columns = disney_df.columns.str.lower()

# 공통 컬럼만 추출 (필요한 컬럼만 남기는 것으로 활용해도 OK)
common_cols = ['show_id', 'type', 'title', 'director', 'cast', 'release_year', 'date_added','listed_in','duration']
netflix_df = netflix_df[common_cols]
amazon_df = amazon_df[common_cols]
disney_df = disney_df[common_cols]

# 출처 플랫폼 컬럼 추가
netflix_df['platform'] = 'Netflix'
amazon_df['platform'] = 'Amazon'
disney_df['platform'] = 'Disney'

amazon_df.head(20)

<ipython-input-2-1a3e094efa08>:13: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  netflix_df['platform'] = 'Netflix'
<ipython-input-2-1a3e094efa08>:14: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  amazon_df['platform'] = 'Amazon'
<ipython-input-2-1a3e094efa08>:15: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/inde

show_id     type                              title  \
0       s1    Movie                The Grand Seduction   
1       s2    Movie               Take Care Good Night   
2       s3    Movie               Secrets of Deception   
3       s4    Movie                 Pink: Staying True   
4       s5    Movie                      Monster Maker   
5       s6    Movie              Living With Dinosaurs   
6       s7    Movie                          Hired Gun   
7       s8    Movie                       Grease Live!   
8       s9    Movie                    Global Meltdown   
9      s10    Movie                     David's Mother   
10     s11    Movie                     Forest Fairies   
11     s12    Movie                          Take Care   
12     s13    Movie           The Night Eats The World   
13     s14    Movie                         Resilencia   
14     s15    Movie  Elon Musk: The Real Life Iron Man   
15     s16    Movie                         Summer '03   
16     s17    Movie                           Zoombies   
17     s18  TV Show                         Zoo Babies   
18     s19  TV Show      Zoë Coombs Marr: Bossy Bottom   
19     s20    Movie                                Zoe   

                       director  \
0                  Don McKellar   
1                  Girish Joshi   
2                   Josh Webber   
3                Sonia Anderson   
4                  Giles Foster   
5                  Paul Weiland   
6                   Fran Strine   
7   Thomas Kail, Alex Rudzinski   
8                 Daniel Gilboy   
9         Robert Allan Ackerman   
10               Justin G. Dyck   
11                 Liz Tuccillo   
12             Dominique Rocher   
13                Jep Barcelona   
14               Sonia Anderson   
15                Becca Gleason   
16                 Glenn Miller   
17                          NaN   
18                          NaN   
19                Drake Doremus   

                                                 cast  release_year  \
0      Brendan Gleeson, Taylor Kitsch, Gordon Pinsent          2014   
1    Mahesh Manjrekar, Abhay Mahajan, Sachin Khedekar          2018   
2   Tom Sizemore, Lorenzo Lamas, Robert LaSardo, R...          2017   
3   Interviews with: Pink, Adele, Beyoncé, Britney...          2014   
4   Harry Dean Stanton, Kieran O'Brien, George Cos...          1989   
5   Gregory Chisholm, Juliet Stevenson, Brian Hens...          1989   
6   Alice Cooper, Liberty DeVitto, Ray Parker Jr.,...          2017   
7   Julianne Hough, Aaron Tveit, Vanessa Hudgens, ...          2016   
8   Michael Paré, Leanne Khol Young, Patrick J. Ma...          2017   
9     Kirstie Alley, Sam Waterston, Stockard Channing          1994   
10  Emily Wilder, Adrian Cowan, Gary Mckenzie, Jer...          2016   
11            Leslie Bibb, Kevin Curtis, Nadia Dajani          2014   
12  Anders Danielsen Lie, Golshifteh Farahani, Den...          2018   
13  Rafinha Alcantara, Marc-André Ter Stegen, Serg...          2020   
14  Elon Musk, Per Wimmer, Julie Anderson-Ankenbra...          2018   
15  Joey King, Jack Kilmer, Andrea Savage, Paul Sc...          2019   
16         Marcus Anderson, Kaiwi Lyman, Andrew Asper          2016   
17                         Narrator - Gillian Barlett          2008   
18                                    Zoë Coombs Marr          2020   
19  Ewan McGregor, Léa Seydoux, Theo James, Mirand...          2018   

        date_added                          listed_in  duration platform  
0   March 30, 2021                      Comedy, Drama   113 min   Amazon  
1   March 30, 2021               Drama, International   110 min   Amazon  
2   March 30, 2021            Action, Drama, Suspense    74 min   Amazon  
3   March 30, 2021                        Documentary    69 min   Amazon  
4   March 30, 2021                     Drama, Fantasy    45 min   Amazon  
5   March 30, 2021                      Fantasy, Kids    52 min   Amazon  
6   March 30, 2021      Documen

#3사 Data 합치기

(세 data set 모두 column 12개로 내용도 동일한 상황)

In [3]:
# 공통 컬럼만 유지하고 행 추가 (inner join)
df = pd.concat([netflix_df, amazon_df, disney_df], join='inner', ignore_index=True)

# platform 컬럼만 랜덤하게 섞어서 3사 데이터가 고루 잘 들어왔는지 확인
df['platform'] = df['platform'].sample(frac=1).reset_index(drop=True)

# 결과 확인
df.head(10)

show_id     type                             title  \
0      s1    Movie              Dick Johnson Is Dead   
1      s2  TV Show                     Blood & Water   
2      s3  TV Show                         Ganglands   
3      s4  TV Show             Jailbirds New Orleans   
4      s5  TV Show                      Kota Factory   
5      s6  TV Show                     Midnight Mass   
6      s7    Movie  My Little Pony: A New Generation   
7      s8    Movie                           Sankofa   
8      s9  TV Show     The Great British Baking Show   
9     s10    Movie                      The Starling   

                        director  \
0                Kirsten Johnson   
1                            NaN   
2                Julien Leclercq   
3                            NaN   
4                            NaN   
5                  Mike Flanagan   
6  Robert Cullen, José Luis Ucha   
7                   Haile Gerima   
8                Andy Devonshire   
9                 Theodore Melfi   

                                                cast  release_year  \
0                                                NaN          2020   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...          2021   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...          2021   
3                                                NaN          2021   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          2021   
5  Kate Siegel, Zach Gilford, Hamish Linklater, H...          2021   
6  Vanessa Hudgens, Kimiko Glenn, James Marsden, ...          2021   
7  Kofi Ghanaba, Oyafunmike Ogunlano, Alexandra D...          1993   
8  Mel Giedroyc, Sue Perkins, Mary Berry, Paul Ho...          2021   
9  Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...          2021   

           date_added                                          listed_in  \
0  September 25, 2021                                      Documentaries   
1  September 24, 2021    International TV Shows, TV Dramas, TV Mysteries   
2  September 24, 2021  Crime TV Shows, International TV Shows, TV Act...   
3  September 24, 2021                             Docuseries, Reality TV   
4  September 24, 2021  International TV Shows, Romantic TV Shows, TV ...   
5  September 24, 2021                 TV Dramas, TV Horror, TV Mysteries   
6  September 24, 2021                           Children & Family Movies   
7  September 24, 2021   Dramas, Independent Movies, International Movies   
8  September 24, 2021                       British TV Shows, Reality TV   
9  September 24, 2021                                   Comedies, Dramas   

    duration platform  
0     90 min   Amazon  
1  2 Seasons   Amazon  
2   1 Season   Disney  
3   1 Season   Amazon  
4  2 Seasons   Amazon  
5   1 Season   Disney  
6     91 min   Amazon  
7    125 min   Amazon  
8  9 Seasons  Netflix  
9    104 min   Amazon

#cast column 분리

In [4]:
# 'cast' 컬럼을 ',' 기준으로 분리해서 리스로 만들기
df['actor'] = df['cast'].apply(lambda x: str(x).split(','))

# # 같은 영화에 출연한 배우(cast 컬럼의 리스트)를 여러 행으로 확장
df_exploded = df.explode('actor')

# # 배우 이름 앞뒤 공백 제거
df_exploded['actor'] = df_exploded['actor'].str.strip()

df_exploded.head(20)

# # 결과 확인 (타이틀과 배우 컬럼만 보기)
df_exploded[['title', 'actor']].head(50)

title                 actor
0   Dick Johnson Is Dead                   nan
1          Blood & Water            Ama Qamata
1          Blood & Water           Khosi Ngema
1          Blood & Water         Gail Mabalane
1          Blood & Water        Thabang Molaba
1          Blood & Water      Dillon Windvogel
1          Blood & Water       Natasha Thahane
1          Blood & Water           Arno Greeff
1          Blood & Water     Xolile Tshabalala
1          Blood & Water       Getmore Sithole
1          Blood & Water        Cindy Mahlangu
1          Blood & Water         Ryle De Morny
1          Blood & Water       Greteli Fincham
1          Blood & Water  Sello Maake Ka-Ncube
1          Blood & Water           Odwa Gwanya
1          Blood & Water        Mekaila Mathys
1          Blood & Water         Sandi Schultz
1          Blood & Water        Duane Williams
1          Blood & Water       Shamilla Miller
1          Blood & Water      Patrick Mofokeng
2              Ganglands         Sami Bouajila
2              Ganglands          Tracy Gotoas
2              Ganglands           Samuel Jouy
2              Ganglands         Nabiha Akkari
2              Ganglands        Sofia Lesaffre
2              Ganglands      Salim Kechiouche
2              Ganglands     Noureddine Farihi
2              Ganglands  Geert Van Rampelberg
2              Ganglands       Bakary Diombera
3  Jailbirds New Orleans                   nan
4           Kota Factory            Mayur More
4           Kota Factory        Jitendra Kumar
4           Kota Factory            Ranjan Raj
4           Kota Factory             Alam Khan
4           Kota Factory         Ahsaas Channa
4           Kota Factory        Revathi Pillai
4           Kota Factory            Urvi Singh
4           Kota Factory            Arun Kumar
5          Midnight Mass           Kate Siegel
5          Midnight Mass          Zach Gilford
5          Midnight Mass      Hamish Linklater
5          Midnight Mass          Henry Thomas
5          Midnight Mass        Kristin Lehman
5          Midnight Mass       Samantha Sloyan
5          Midnight Mass           Igby Rigney
5          Midnight Mass           Rahul Kohli
5          Midnight Mass        Annarah Cymone
5          Midnight Mass         Annabeth Gish
5          Midnight Mass            Alex Essoe
5          Midnight Mass          Rahul Abburi

In [12]:
df = pd.concat([netflix_df[['title', 'cast', 'listed_in']],
                amazon_df[['title', 'cast', 'listed_in']],
                disney_df[['title', 'cast', 'listed_in']]],
               join='inner', ignore_index=True)

In [13]:
# listed_in 컬럼을 쉼표 기준으로 나누기
df['genre_list'] = df['listed_in'].str.split(', ')

# title을 유지하면서 genre_list를 여러 행으로 확장
df_exploded_genre = df[['title', 'genre_list']].explode('genre_list')

# 결과 확인
df_exploded_genre.head(10)

title              genre_list
0   Dick Johnson Is Dead           Documentaries
1          Blood & Water  International TV Shows
1          Blood & Water               TV Dramas
1          Blood & Water            TV Mysteries
2              Ganglands          Crime TV Shows
2              Ganglands  International TV Shows
2              Ganglands   TV Action & Adventure
3  Jailbirds New Orleans              Docuseries
3  Jailbirds New Orleans              Reality TV
4           Kota Factory  International TV Shows

In [14]:
unique_genres = df_exploded_genre['genre_list'].nunique()
print(f"현재 데이터셋 내 고유한 장르 개수: {unique_genres}")

현재 데이터셋 내 고유한 장르 개수: 102


In [18]:
genre_counts = df_exploded_genre['genre_list'].value_counts()
print(genre_counts.head(51))
print(genre_counts.tail(51))

genre_list
Drama                        3821
International Movies         2752
Comedy                       2625
Dramas                       2427
Comedies                     1674
Action                       1657
Suspense                     1501
International TV Shows       1351
Kids                         1226
Documentary                  1167
Animation                    1089
Special Interest              980
Horror                        875
Documentaries                 869
Action & Adventure            859
TV Dramas                     763
Independent Movies            756
Romance                       694
Children & Family Movies      641
Family                        632
Romantic Movies               616
TV Comedies                   581
Thrillers                     577
Science Fiction               548
Docuseries                    517
and Culture                   483
Entertainment                 483
Arts                          483
Crime TV Shows                470
Act

In [20]:
# 'Unscripted'가 실제 데이터에 존재하는지 확인
print(df['listed_in'].unique())  # 고유한 장르 리스트 출력

# 'Unscripted'가 있는 행 출력
print(df[df['listed_in'].str.contains('Unscripted', na=False)])

['Documentaries' 'International TV Shows, TV Dramas, TV Mysteries'
 'Crime TV Shows, International TV Shows, TV Action & Adventure' ...
 'Animation, Family, Mystery' 'Biographical, Comedy, Drama'
 'Action-Adventure, Animals & Nature, Animation']
                                           title  \
8871                         Would I Lie to You?   
8873   World’s Toughest Race: Eco-Challenge Fiji   
8893                              Wipeout Canada   
8894                           Wipeout Australia   
8936                             What's My Line?   
...                                          ...   
17808                            Roadkill Garage   
17977                                Judge Faith   
18125               Landscape Artist of the Year   
18255           Infinity Paranormal Investigate.   
18344                                Pan y Circo   

                                                    cast  \
8871                David Mitchell, Lee Mack, Rob Brydon   
8873     

#cast column과 listed_in column 따로 빼기 + 결측치 채우기

- cast column에는 3가지 data set 모두 약 10% 정도의 null값 존재

- listed_in column에는 null값 없음

- 별점에 대한 imdb data set은 나중에 다시 불러와서 비교분석할 예정

In [23]:
df = df.copy()  # 혹시 모를 경고 방지를 위해 복사본 생성
df['cast'] = df['cast'].fillna('Unknown')

df['cast'].isnull().sum()

0

In [24]:
genre_mapping = {
    'Action & Adventure': ["Action", "Action & Adventure", "Action-Adventure", "TV Action & Adventure"],
    'War': ["War", "Military and War"],
    'Talk Shows & Variety': ["Talk Show", "Talk Show and Variety"],
    'Biographical': ["Biographical"],
    'Arts & Culture': ["and Culture", "Arts"],
    'Action': ["Action", "Action-Adventure", "TV Action & Adventure"],
    'Mystery & Suspense': ["Mystery", "Suspense"],
    'Musical': ["Musical"],
    'Biography': ["Biography", "Biographical"],
    'Romance': ["Romance", "Romantic Movies", "Romantic TV Shows", "Romantic Comedy"],
    'Sports': ["Sports", "Sports Movies"],
    'Animation': ["Animation", "Anime", "Anime Series"],
    'History': ["History", "Historical"],
    'Historical': ["History", "Historical"],
    'Sport': ["Sports", "Sports Movies"],
    'Western': ["Western"],
    'Kids & Family': ["Kids", "Children & Family Movies", "Kids' TV", "Family"],
    'British': ["British TV Shows"],
    'News': ["News"],
    'Mystery': ["Mystery", "TV Mysteries"],
    'Sci-Fi & Fantasy': ["Science Fiction", "Sci-Fi & Fantasy", "TV Sci-Fi & Fantasy"],
    'Music & Musicals': ["Music & Musicals", "Music Videos and Concerts", "Concert Film"],
    'Family': ["Family", "Kids", "Children & Family Movies", "Kids' TV"],
    'Reality_TV': ["Reality TV", "Unscripted", "Reality"],
    'Crime & Thriller': ["Crime", "Crime TV Shows", "Police/Cop", "Thrillers"],
    'Thriller': ["Thriller", "Suspense"],
    'Coming of Age': ["Coming of Age", "Teen TV Shows"],
    'Documentary': ["Documentary", "Docuseries", "Documentaries", "Science & Nature TV"],
    'Suspense & Thriller': ["Suspense", "Thriller"],
    'Fantasy': ["Fantasy", "Sci-Fi & Fantasy", "TV Sci-Fi & Fantasy"],
    'Superhero': ["Superhero"],
    'Animals & Nature': ["Animals & Nature"],
    'Teen': ["Teen", "Teen TV Shows"],
    'Korean': ["Korean TV Shows"],
    'Sci-Fi': ["Science Fiction", "Sci-Fi & Fantasy"],
    'Game-Show': ["Game Show / Competition"],
    'Talk-Show': ["Talk Show", "Talk Show and Variety"],
    'Drama': ["Drama", "Dramas", "TV Dramas"],
    'International': ["International Movies", "International TV Shows", "International"],
    'Reality TV': ["Reality TV", "Unscripted", "Reality"],
    'Film-Noir': ["Classic Movies", "Classic & Cult TV"],
    'Music': ["Music", "Music Videos and Concerts", "Concert Film"],
    'Stand-Up Comedy': ["Stand-Up Comedy", "Stand-Up Comedy & Talk Shows"],
    'Cult & Classic': ["Cult Movies", "Classic & Cult TV"],
    'Science & Nature': ["Science & Nature TV", "Science & Nature"],
    'War & Military': ["Military and War", "War"],
    'Spanish-Language': ["Spanish-Language TV Shows"],
    'Adventure': ["Adventure", "Action & Adventure"],
    'Anthology': ["Anthology", "Classic & Cult TV"],
    'Crime': ["Crime", "Crime TV Shows", "Police/Cop"],
    'Comedy': ["Comedy", "Comedies", "TV Comedies", "Stand-Up Comedy"],
    'Unscripted': ["Unscripted", "Reality"],
    'Spy/Espionage': ["Spy/Espionage"],
    'Short': ["Short"],
    'Concerts & Performances': ["Music Videos and Concerts", "Concert Film"],
    'Faith & Spirituality': ["Faith & Spirituality"],
    'Fitness & Lifestyle': ["Fitness"],
    'Horror': ["Horror", "Horror Movies", "TV Horror"]
}

In [31]:
import pandas as pd
import re
import numpy as np

# 3개 데이터셋 합치기
df = pd.concat([netflix_df, amazon_df, disney_df], ignore_index=True)

# 필요한 컬럼 선택
df = df[['title', 'platform', 'listed_in', 'cast']]

# 중복값 제거 (drop_duplicates 실행 전에 컬럼들을 문자열로 변환)
df['title'] = df['title'].astype(str)
df['platform'] = df['platform'].astype(str)
df['listed_in'] = df['listed_in'].astype(str)
df['cast'] = df['cast'].astype(str)
df.drop_duplicates(inplace=True)

# title, listed_in, platform 컬럼을 리스트 형식으로 변환
df["title"] = df["title"].fillna("").apply(lambda x: x.split(", ") if x else [])
df["listed_in"] = df["listed_in"].fillna("").apply(lambda x: x.split(", ") if x else [])
df["platform"] = df["platform"].fillna("").apply(lambda x: x.split(", ") if x else [])

# cast 컬럼은 기존 리스트 형태 유지 (문자열로 저장되었을 경우 변환)
# cast 컬럼의 데이터 타입을 문자열로 변환
df['cast'] = df['cast'].astype(str)
# 'cast' 컬럼에서 NaN 값을 'unknown'으로 변환
df['cast'] = df['cast'].apply(lambda x: ['unknown'] if x == 'nan' else eval(x) if x.startswith("[") else x.split(", "))


# 알파벳(A-Z, a-z) 및 숫자(0-9)만 남기고 나머지 문자 제거
def clean_text_list(text_list):
    return ["".join(re.findall(r"[A-Za-z0-9\s]", text)).strip() for text in text_list if text]

df["title"] = df["title"].apply(clean_text_list)
df["listed_in"] = df["listed_in"].apply(clean_text_list)
df["platform"] = df["platform"].apply(clean_text_list)
df["cast"] = df["cast"].apply(clean_text_list)

# title, listed_in, platform 컬럼에서 비어있는 리스트([])가 있는 행 삭제
# cast 컬럼은 이미 null 값이 'unknown'으로 채워져 있으므로 제외
df_cleaned = df[(df["title"].apply(len) > 0) & (df["listed_in"].apply(len) > 0) & (df["platform"].apply(len) > 0)]

# 결과 확인
df_cleaned.head()

title   platform  \
0   [Dick Johnson Is Dead]  [Netflix]   
1           [Blood  Water]  [Netflix]   
2              [Ganglands]  [Netflix]   
3  [Jailbirds New Orleans]  [Netflix]   
4           [Kota Factory]  [Netflix]   

                                           listed_in  \
0                                    [Documentaries]   
1  [International TV Shows, TV Dramas, TV Mysteries]   
2  [Crime TV Shows, International TV Shows, TV Ac...   
3                           [Docuseries, Reality TV]   
4  [International TV Shows, Romantic TV Shows, TV...   

                                                cast  
0                                          [unknown]  
1  [Ama Qamata, Khosi Ngema, Gail Mabalane, Thaba...  
2  [Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nab...  
3                                          [unknown]  
4  [Mayur More, Jitendra Kumar, Ranjan Raj, Alam ...

No charts were generated by quickchart


In [25]:
import pandas as pd

# 예시 데이터프레임
data = {
    'title': ['Movie A', 'Movie B'],
    'cast': [['Actor 1', 'Actor 2'], ['Actor 3', 'Actor 4']],
    'listed_in': [['Action', 'Comedy'], ['Drama', 'Horror']]
}
df = pd.DataFrame(data)

# 1. 'cast' 컬럼을 개별 배우로 쪼개고, 'listed_in'을 58개 카테고리에 매핑
expanded_rows = []

for _, row in df.iterrows():
    title = row['title']
    actors = row['cast']
    genres = row['listed_in']

    # 각 배우와 영화에 대해, 장르를 반복하여 새로운 행을 생성
    for actor in actors:
        for genre in genres:
            # 장르를 카테고리에 맞게 변경
            for category, subgenres in genre_mapping.items():
                if genre in subgenres:
                    expanded_rows.append({'title': title, 'actor': actor, 'genre_category': category})

# 새로운 데이터프레임 생성
expanded_df = pd.DataFrame(expanded_rows)

# 결과 확인
print(expanded_df)

     title    actor      genre_category
0  Movie A  Actor 1  Action & Adventure
1  Movie A  Actor 1              Action
2  Movie A  Actor 1              Comedy
3  Movie A  Actor 2  Action & Adventure
4  Movie A  Actor 2              Action
5  Movie A  Actor 2              Comedy
6  Movie B  Actor 3               Drama
7  Movie B  Actor 3              Horror
8  Movie B  Actor 4               Drama
9  Movie B  Actor 4              Horror
